src/test/groovy/graphql/execution/reactive/ReactiveSupportTest.groovy

```diff
@@ -0,0 +1,222 @@
+
+        when:
+        CompletableFuture<?> cf = ReactiveSupport.fetchedObject(reactiveObject) as CompletableFuture<?>
+
+        then:
+        !cf.isCancelled()
+        !cf.isCompletedExceptionally()
+        !cf.isDone()
+
+        when:
+        def cfCancelled = cf.cancel(true)
+
+        then:
+        cfCancelled
+        cf.isCancelled()
+        cf.isCompletedExceptionally()
+        cf.isDone()
+
+        when:
+        cf.join()
+
+        then:
+        thrown(CancellationException.class)
+
+        where:
+        reactiveObject                 || _
+        delayedMono("X", 1000)         || _
+        toFlow(delayedMono("X", 1000)) || _
+
+    }
+
+
+    def "can get a reactive Flux and only take one value and make a CF from it"() {
+
+        def xyzStrings = ["X", "Y", "Z"]
+        when:
+        def countingFlux = new CountingFlux(xyzStrings)
+        CompletableFuture<?> cf = ReactiveSupport.fetchedObject(countingFlux.flux) as CompletableFuture<?>
+
+        then:
+        !cf.isCancelled()
+        !cf.isCompletedExceptionally()
+        cf.isDone()
+
+        cf.join() == "X"
+        countingFlux.count == 1
+
+        when:
+        def capturingSubscriber = new CapturingSubscriber<>()
+        countingFlux.flux.subscribe(capturingSubscriber)
+
+        then:
+        // second subscriber
+        capturingSubscriber.events == ["X", "Y", "Z"]
+        countingFlux.count == 4
+    }
+
+    def "integration test showing reactive values in data fetchers as well as the ones we know and love"() {
+        def sdl = """
+            type Query {
+                reactorField : String
+                flowField : String
+                cfField : String
+                materialisedField : String
+            }
+        """
+
+        // with some delay
+        def reactorDF = { env -> delayedMono("reactor", 100) } as DataFetcher
+        def flowDF = { env -> toFlow(delayedMono("flow", 50)) } as DataFetcher
+
+        def cfDF = { env -> CompletableFuture.completedFuture("cf") } as DataFetcher
+        def materialisedDF = { env -> "materialised" } as DataFetcher
+
+        def schema = TestUtil.schema(sdl, [Query: [
+                reactorField     : reactorDF,
+                flowField        : flowDF,
+                cfField          : cfDF,
+                materialisedField: materialisedDF,
+        ]])
+        def graphQL = GraphQL.newGraphQL(schema).build()
+
+        when:
+        def er = graphQL.execute("""
+            query q {
+                reactorField
+                flowField
+                cfField
+                materialisedField
+            }
+        """)
+
+        then:
+        er.errors.isEmpty()
+        er.data == [
+                reactorField     : "reactor",
+                flowField        : "flow",
+                cfField          : "cf",
+                materialisedField: "materialised"
+        ]
+    }
```

Comment:
full integration tests showing it working.

We know CF error handling works already so there is no need for integration tests showing all that